### ISSUE TITLE
Signing key installation under WSL Ubuntu can fail due to gpg bug

### ISSUE THREAD

**Issue by simonbrady** [Azure/azure-docs-cli-python#1164]
The advice under _Troubleshooting / apt-key fails with "No dirmngr"_ is to install the dirmngr package. However, in Ubuntu under WSL this error can be triggered even with dirmngr installed, due to https://github.com/Microsoft/WSL/issues/3286. The workaround is to follow the advice in that issue and import the key this way instead:

`curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xBC528686B50D79E339D3721CEB3E94ADBE1229CF" | sudo apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg add`

Can this please be added to the docs?

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe1526a9-7f22-504e-484c-0f3edc11b178
* Version Independent ID: c5dec108-efcd-aa77-04b5-803b6b894dba
* Content: [Install the Azure CLI on Linux with apt](https://docs.microsoft.com/en-gb/cli/azure/install-azure-cli-apt?view=azure-cli-latest)
* Content Source: [docs-ref-conceptual/install-azure-cli-apt.md](https://github.com/Azure/azure-docs-cli-python/blob/master/docs-ref-conceptual/install-azure-cli-apt.md)
* Product: **azure**
* GitHub Login: @sptramer
* Microsoft Alias: **sttramer**

**Comment by LyalinDotCom** [comment]
This advice really helped get me unblocked so thank you @simonbrady! I then ran both 
sudo apt-get update
and
sudo apt-get install azure-cli

to get the CLI installed. I am new to Linux subsystem so without your help I would have been totally blocked

**Comment by LyalinDotCom** [comment]
Seems the issue is more complicated then it seems with this doc. I reset my Ubuntu environment to try again from scratch and following my own steps above didnt work. I had to run various command in various orders from both this suggested fix and the original commands from the top (line by line instead of copy all, somehow that might have made a difference) and after enough time mocking with this it finally worked.

I am now stuck on a new issue with unpacking azure-cli taking forever (running for 4 minutes now)

There is a good thread about this issue here: https://github.com/Microsoft/WSL/issues/2775

I am hoping it completes eventually so going to let it run now as some have suggested, but us not giving people the ability to see unpacking progress is likely the real problem as the last post on this thread indicates.

**Comment by benc-uk** [comment]
When the new 2.0 CLI first came out it used the existing Python runtime on your system. At some point they switched and now it includes it's own "private" install of Python and all the libraries, this is a LOT of files, and the reason for the crazy unpacking time. 10~20 mins is normal, and you have to go through it every time you update the CLI.

I've seen several other people assume the install has hung too.  
I think the user experience here could be improved a lot.

**Comment by sptramer** [comment]
@benc-uk - I've noticed this problem when I checked the `.deb` requirements information for the provided package, which actually doesn't include the appropriate prereqs such as openssl or libffi. If you would [file an engineering bug](https://github.com/azure/azure-cli/issues) I'm sure it would be appreciated. Link back to this ticket as user feedback.

@simonbrady - Since this is fixed in Win 10 1809, the troubleshooting instructions won't be updated at this time. Hopefully the WSL team has fixed the IPC/FFI call issues for good.
